apollo-compiler/src/main/kotlin/com/apollographql/android/compiler/JavaTypeResolver.kt

```diff
@@ -31,6 +32,7 @@ class JavaTypeResolver(
       when (context.nullableValueType) {
         NullableValueType.APOLLO_OPTIONAL -> parameterizedOptional(javaType)
         NullableValueType.GUAVA_OPTIONAL -> parameterizedGuavaOptional(javaType)
+        NullableValueType.JAVA8_OPTIONAL -> parameterizedJava8Optional(javaType)
```

Comment:
Nitpick I think you can call if just java optional so we don't need to change anything once java 9 comes out